internal/app/go-filecoin/connectors/storage_market/client.go

```diff
@@ -58,8 +59,8 @@ func (s *StorageClientNodeConnector) AddFunds(ctx context.Context, addr address.
 }
 
 // EnsureFunds checks the current balance for an address and adds funds if the balance is below the given amount
-func (s *StorageClientNodeConnector) EnsureFunds(ctx context.Context, addr, walletAddr address.Address, amount abi.TokenAmount) error {
-	balance, err := s.GetBalance(ctx, addr)
+func (s *StorageClientNodeConnector) EnsureFunds(ctx context.Context, addr, walletAddr address.Address, amount abi.TokenAmount, tok shared.TipSetToken) error {
+	balance, err := s.GetBalance(ctx, addr, tok)
```

Comment:
It seems to me that we should be deserializing the toks (which I think are just cbor bytes of a tipset key) as soon as they cross into the go-filecoin part of the code base, rather than letting the token move between function calls internal to the connector.

I can see this case might be unavoidable because GetBalance is on the API boundary.  However the internal `listDeals` shouldn't be.

Reply:
Makes sense!